src/rgw/rgw_rest_s3.cc

```diff
@@ -1384,10 +1384,14 @@ void RGWPutObj_ObjStore_S3::send_response()
       dump_errno(s);
       dump_etag(s, etag);
       dump_content_length(s, 0);
+      if (!version_id.empty())
+        dump_header(s, "x-amz-version-id", version_id.c_str());
```

Comment:
there's a helper function to make these a bit easier:
```c++
      dump_header_if_nonempty(s, "x-amz-version-id", version_id);
```

Reply:
@cbodley Thanks for your advice. I have amended it.